File: Data Structures & Algorithms/insert-new-interval/submission-2.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> res;
        if(intervals.empty()){
            return {newInterval};
        }
        int n= intervals.size();
        int target= newInterval[0];
        int left =0, right=n-1;
        int mid;
        while(left<=right){
            mid= (left+right)/2;
            if(intervals[mid][0]<target){
                left= mid+1;
            }
            else{
                right= mid-1;
            }
        }
        intervals.insert(intervals.begin()+left, newInterval);
        for(const auto& interval:intervals){
            if(res.empty()|| res.back()[1]< interval[0]){
                res.push_back(interval);
            }
            else {
                res.back()[1]=max(res.back()[1], interval[1]);
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l=1;
        r= max(piles)
        ans= r
        while l<=r :
            mid= l+(r-l)//2
            totalbanana=0
            for p in piles:
                totalbanana += (p+mid-1)//mid
            if totalbanana<= h:
                ans= mid
                r= mid-1
            else:
                l=mid+1
        return ans